backend.go

```diff
@@ -23,11 +24,77 @@ type jsonFileBackend struct {
 	filename string
 }
 
+type flagJson struct {
+	Name   string
+	Active bool
+	Rate   float64
+	Rules  []RuleInfo
+}
+
+type ruleInfoJson struct {
+	Type    string     `json:"type"`
+	OnMatch RuleAction `json:"on_match"`
+	OnMiss  RuleAction `json:"on_miss"`
+}
+
 type JSONFormat struct {
 	Flags       []Flag  `json:"flags"`
 	UpdatedTime float64 `json:"updated"`
 }
 
+func (ri *Flag) UnmarshalJSON(buf []byte) error {
+	var raw flagJson
+	err := json.Unmarshal(buf, &raw)
+	if err != nil {
+		return err
+	}
+
+	if len(raw.Rules) == 0 {
+		if raw.Rate > 0 {
+			raw.Active = true
```

Comment:
And if the Rate is 0, then we're still ok, because that's equivalent to inactive. Cooool. Again, comment this :)

Reply:
yup! i added a comment here as well.